CurrencyControl/Themes/Generic.xaml

```diff
@@ -0,0 +1,55 @@
+﻿<ResourceDictionary
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="clr-namespace:CurrencyControl">
+    <Style x:Key="txtStyle"  TargetType="TextBox" >
+        <Setter  Property="Background" Value="{Binding CustomBackground, RelativeSource={RelativeSource AncestorType={x:Type local:CurrencyControl}}}"/>
+        <Setter  Property="Foreground" Value="{Binding CustomForeground, RelativeSource={RelativeSource AncestorType={x:Type local:CurrencyControl}}}"/>
+        <Setter  Property="FontFamily" Value="{Binding CustomFont, RelativeSource={RelativeSource AncestorType={x:Type local:CurrencyControl}}}"/>
+    </Style>
+    <Style TargetType="{x:Type local:CurrencyControl}">
```

Comment:
According to this XAML, I see that you need to read some basic documentation about WPF control authoring. How styles work, how properties get their values, etc. 
To better understand the problem, please place your ConverterControl to new window in VisualStudio and try to change it's Foreground property to Red in the property grid. You will see that all labels will be always blue, because you hard coded this color, and TextBoxes also won't inherit Red color. What we need from control is to honor Foreground, Background, BorderBrush and BorderThickness properties if application author set them directly or by creating new style for CurrencyConverter control (not for TextBox).
It must be complicated if you didn't work with xaml before. So, please start with playing with your control as if you want to use it in your application and change colors. Change one thing on application side. Then try to find why this change is not applied. Then search how to make it working. Please don't rush, it's very important to get clear understanding of how it all works together

Reply:
I have made some changes in the code based on your feedback. Waiting for your comment. thanks